File: 13-roman-to-integer/roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> map = new HashMap<>();
        int number = 0;

        map.put('I', 1);
        map.put('V', 5);
        map.put('X', 10);
        map.put('L', 50);
        map.put('C', 100);
        map.put('D', 500);
        map.put('M', 1000);

        System.out.println(map.get(s.charAt(0)));

        for (int i = 0; i < s.length(); i++) {
            if (i < s.length() - 1 && map.get(s.charAt(i)) < map.get(s.charAt(i + 1))) {
                number -= map.get(s.charAt(i));
            } else {
                number += map.get(s.charAt(i));
            }
        }

        return number;
    }
}

File: 13-roman-to-integer/roman-to-integer.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function (s) {
    let n = s.length;
    let mp = new Map();
    mp.set("I", 1);
    mp.set("V", 5);
    mp.set("X", 10);
    mp.set("L", 50);
    mp.set("C", 100);
    mp.set("D", 500);
    mp.set("M", 1000);

    let sum = 0;

    for (let i = n - 1; i >= 0; i--) {
        if (s[i] == "V" || s[i] == "X") {
            if (s[i - 1] === "I") {
                sum = sum + mp.get(s[i]) - mp.get(s[i - 1])
                i--;
            } else {
                sum = sum + mp.get(s[i]);
            }
        } else if (s[i] == "L" || s[i] == "C") {
            if (s[i - 1] === "X") {
                sum = sum + mp.get(s[i]) - mp.get(s[i - 1])
                i--;
            } else {
                sum = sum + mp.get(s[i]);
            }
        } else if (s[i] == "D" || s[i] == "M") {
            if (s[i - 1] === "C") {
                sum = sum + mp.get(s[i]) - mp.get(s[i - 1]);
                i--;
            } else {
                sum = sum + mp.get(s[i]);
            }
        } else {
            sum = sum + mp.get(s[i]);
        }
    }

    return sum;
};